metrics-prometheus/src/main/java/com/alibaba/metrics/prometheus/DubboMetricsExports.java

```diff
@@ -0,0 +1,187 @@
+        } else if (o instanceof Boolean) {
+            value = ((Boolean) o) ? 1: 0;
+        } else {
+            return null;
+        }
+        MetricFamilySamples.Sample sample = sampleBuilder.createSample(metricName, "", new ArrayList<String>(), new ArrayList<String>(), value);
+        return new MetricFamilySamples(sample.name, Type.GAUGE, getHelpMessage(metricName.getKey(), gauge), Arrays.asList
+                (sample));
+    }
+
+    public MetricFamilySamples fromTimer(MetricName metricName, Timer timer) {
+        return fromSnapshotAndCount(metricName, timer.getSnapshot(), timer.getCount(),
+                1.0D / TimeUnit.SECONDS.toNanos(1L), getHelpMessage(metricName.getKey(), timer));
+    }
+
+    public MetricFamilySamples fromMeter(MetricName metricName, Meter meter) {
+        MetricFamilySamples.Sample sample = sampleBuilder.createSample(metricName, "_total", new ArrayList<String>(), new ArrayList<String>(), meter.getCount());
+        return new MetricFamilySamples(sample.name, Type.COUNTER, getHelpMessage(metricName.getKey(), meter), Arrays.asList
+                (sample));
+    }
+
+    public MetricFamilySamples fromHistogram(MetricName metricName, Histogram histogram) {
+        return fromSnapshotAndCount(metricName, histogram.getSnapshot(), histogram.getCount(), 1.0,
+                getHelpMessage(metricName.getKey(), histogram));
+    }
+
+    public MetricFamilySamples fromClusterHistogram(MetricName metricName, ClusterHistogram clusterHistogram) {
+        List<MetricFamilySamples.Sample> samples = new ArrayList<MetricFamilySamples.Sample>();
+        Map<Long, Map<Long, Long>> buckets = clusterHistogram.getBucketValues(0L);
+        for (Map.Entry<Long, Map<Long, Long>> entry : buckets.entrySet()) {
+            String suffix = "_time" + entry.getKey().toString();
+            Map<Long, Long> bucket = entry.getValue();
+            for (Map.Entry<Long, Long> entry1 : bucket.entrySet()) {
+                samples.add(sampleBuilder.createSample(metricName, suffix, Arrays.asList("bucket"), Arrays.asList(entry1.getKey().toString()), entry1.getValue()));
+            }
+        }
+        return new MetricFamilySamples(samples.get(0).name, Type.SUMMARY, getHelpMessage(metricName.getKey(), clusterHistogram), samples);
+    }
+
+    public MetricFamilySamples fromCompass(MetricName metricName, Compass compass) {
+        long lastUpdateTime = compass.lastUpdateTime();
+        int bucketInterval = compass.getInstantCountInterval();
+        long start = getNormalizedStartTime(lastUpdateTime, bucketInterval);
+        BucketCounter successCounter = compass.getBucketSuccessCount();
+        List<MetricFamilySamples.Sample> samples = new ArrayList<MetricFamilySamples.Sample>();
+        for (Map.Entry<String, BucketCounter> entry : compass.getErrorCodeCounts().entrySet()) {
+            String tag = entry.getKey();
+            long count = entry.getValue().getBucketCounts().get(start);
```

Comment:
`entry.getValue().getBucketCounts().get(start)` may return a null value, need to check it.